src/builder/modules/Generator_Builder.res

```diff
@@ -437,6 +437,48 @@ let fillBackgroundColor = (model: Model.t, fillStyle: string) => {
   }
 }
 
+let drawLine = (
+  model: Model.t,
+  (dx, dy, dw, dh): rectangle,
+  ~color: string,
+  ~width: float,
+  ~pattern: array<int>,
+  ~offset: int,
+) => {
+  let x = Belt.Int.toFloat(dx)
+  let y = Belt.Int.toFloat(dy)
+  let w = Belt.Int.toFloat(dw)
+  let h = Belt.Int.toFloat(dh)
+
+  let angle = w === 0.0 ? Js.Math._PI /. 2.0 : Js.Math.atan2(~y=h, ~x=w, ())
+  let ow = Js.Math.sin(angle) *. 0.5
+  let oh = Js.Math.cos(angle) *. 0.5
```

Comment:
I didn't quite follow what this bit of code was doing, but I think if we switch to `from` and `to` positions, then we won't need to manipulate the values?

I'm assuming we can just do this:

```
let (x1, y1) = from
let (x2, y2) = to

context->Context2d.moveTo(x1, y1)
context->Context2d.lineTo(x2, y2)
```
Is that right?

Reply:
This code is so that the line is offset perpendicular to its slope. When starting at integer positions, the line ends up between two pixel rows and shows up transparent and on both rows, instead of fully opaque on one row. This code creates an offset so that the line can be drawn properly.